src/client/Microsoft.Identity.Client/Utils/JsonHelper.cs

```diff
@@ -155,6 +157,125 @@ internal static long ExtractParsedIntOrZero(JObject json, string key)
         internal static bool TryGetValue(JsonObject json, string propertyName, out JsonNode value) => json.TryGetPropertyValue(propertyName, out value);
 
         internal static T GetValue<T>(JsonNode json) => json != null ? json.GetValue<T>() : default;
+
+        /// <summary>
+        /// Combines two JSON strings into a single JSON string with indentation.
+        /// </summary>
+        /// <param name="originalJson">The original JSON string to merge.</param>
+        /// <param name="newContent">The additional JSON content to merge.</param>
+        /// <returns>A UTF-8 encoded string representing the merged JSON.</returns>
+        /// Code reference : https://github.com/dotnet/runtime/issues/31433
+        internal static string Merge(string originalJson, string newContent)
```

Comment:
My first thought was that this creates quite a few allocations. We start with 2 JsonObjects, which are converted to string, then parsed to JsonDocument, then outputBuffer is allocated, which at the end is written to string, and then parsed to JsonObject, then converted to string again. So maybe this can be made more efficient by updating one of the JsonObject s in place or at the very least updating the parameters and return values of these methods so we don't unnecessarily convert to string and back.

Now, this isn't a hot path and the JSON is expected to be small, so not that big of a deal to keep as-as, but we should do simple improvements if possible.

Reply:
agree that JSON would be small and not sure if there is any thing much we can improve, code seems to be well structured, just made sure to add a good test coverage so all paths are tested

Reply:
@pmaytak updated please review